### PR TITLE
Remic/tuto1

### PR DESCRIPTION
**Pull request by remicousin**
Making a tutorial that builds the ENACTS Monthly Maproom.

I won't be able to wait for approvals and merge and open PR again but I'll take any feedback and I'll have to figure out what I can afford to implement or not.

It all starts with running maproom01.py with the corresponding 01 config file and you keep going. You need not have done one to be able to do the next one. But the next one includes the work expected to have been done in the previous one.

I am adding everybody here since we all either will need to teach, or learn, and likely both for most of us.

### CODE REVIEW

**Comment by remicousin** [comment]
And don't be shy in proposing the impossible: I'll be own judge of what I can implement here. Good ideas will be made issues for future trainings.

**Comment by aaron-kaplan** [comment]
Could you provide a summary of what new concepts and functionality are introduced in each step?

**Comment by remicousin** [comment]
Did you mean to put that summary some place in particular?

**Comment by aaron-kaplan** [comment]
A top-level README could be a good place for it?

**Comment by aaron-kaplan** [comment]
I thought you were going to present the steps as git commits as you did in maproom_devtuto, rather than numbered versions that are all present as once. Do you intend to keep the current structure, or is it temporary?

**Comment by remicousin** [comment]
> A top-level README could be a good place for it?

**Comment by remicousin** [comment]
In the uicore tuto, it's easy to line up the lessons with the tab structure for the Maproom text. So that one can go from one Lesson's instruction to the next. Here I am not sure... I could get some collapsing texts...? I am going to try that.

**Comment by remicousin** [comment]
An idea how I can have piece of code formatted inside an html.P

**Comment by remicousin** [comment]
sorry, I must have clicked buttons by mistake. I found Details to collapse text easily.

**Comment by jturmelle** [comment]
A combination of the <code> tag and css should acheive the affect you want?

https://www.w3schools.com/tags/tag_code.asp <https://www.w3schools.com/tags/tag_code.asp>

J

—
Jeff Turmelle (he/him)
Lead Systems Analyst
International Research Institute for Climate and Society <http://iri.columbia.edu/>
Columbia Climate School <https://climate.columbia.edu/>
(845) 652-3461

File: maproom.py [hunk 1]
```diff
@@ -61,11 +61,31 @@ def data_tiles(tz, tx, ty, variable):
...
-    
-    
+
+
+# Lesson 8 starts
+@APP.callback(
+    Output("colorbar", "colorscale"),
+    Output("colorbar", "min"),
+    Output("colorbar", "max"),
+    Input("variable", "value"),
```
Comment (drewmresnick) [inline maproom.py]: Here you do not use variable input to select a variable from within a ds. If you are using just a single data variable then maybe there is no use for having this as a callback. But if you want to update the colorbar according to different variable data, refer to the [most recent commit I pushed](https://github.com/iridl/lms-monthly/pull/5/commits/16c40887b2cfb9b38395013d3283daac47ca3b52)

Comment (remicousin) [reply maproom.py]: The 2nd variable is coming a bit later in the Tuto. Could probably have done this a different way...

**Comment by remicousin** [comment]
ok. I am going to make a pause here. I "saved" in the "expected" folder a possible end result of where one could get running the Tutorial; and I reverted the actual Tutorial files back to what their starting point should be (should probably think of how to manage that better, and whether at the end I should keep an "expected" results or not...). So you all can play with it and comment about it if you feel like it and have the time. The degree of help there is in each Lesson is likely very variable, but for now, this is meant to help me run the show... more than making sure it is stand-alone.

I still want to, probably in that order (some of these might turn to be several "Lessons"):

1. cover the temperature colorscale ticks case
2. compute climatology
3. map a chosen month of climatology
4. callback on map title
5. add a maker that responds to click and type-in boxes (ui_component)
6. have a local plot respond to marker change (first precip)
7. have the local plot respond to variable as well
8. overlay admin boundaries
9. mouse hover give admin boundaries label/name
10. Anything else that seems basic and important?

I might not be able to make all those actual Lessons before D-day... we'll see. I rarely managed to work in planes even when I wanted to, but with 20+ hours of travel, it might happen. I also have all Monday off.

As I might not have mentioned: feedbacks and comments welcome.

(Also, I likely diverted from Drew is doing in lms-monthly but I had to go with what I know how to do -- I'm likely going to pick up the local graphs from there, though).

**Pull request merged**